pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -337,6 +342,67 @@ type GettableApiAlertingConfig struct {
 	Receivers []*GettableApiReceiver `yaml:"receivers,omitempty" json:"receivers,omitempty"`
 }
 
+func getMissingJSONTags(t reflect.Type, m map[string]string) {
+	if t.Kind() != reflect.Struct {
+		return
+	}
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		jsonTag := f.Tag.Get("json")
+		if jsonTag == "" {
+			if yamlTag, ok := t.Field(i).Tag.Lookup("yaml"); ok {
+				// TODO check for conflicts
+				m[f.Name] = strings.Split(yamlTag, ",")[0]
+			}
+		}
+		switch t.Field(i).Type.Kind() {
+		case reflect.Struct:
+			getMissingJSONTags(t.Field(i).Type, m)
+		case reflect.Ptr:
+			getMissingJSONTags(t.Field(i).Type.Elem(), m)
+		}
+	}
+}
+
+func (c *GettableApiAlertingConfig) MarshalJSON() ([]byte, error) {
+	type cloneGettableApiAlertingConfig struct {
+		Config `yaml:",inline"`
+
+		// Override with our superset receiver type
+		Receivers []*GettableApiReceiver `yaml:"receivers,omitempty" json:"receivers,omitempty"`
+	}
+
+	clone := cloneGettableApiAlertingConfig{
+		Config:    c.Config,
+		Receivers: c.Receivers,
+	}
+
+	enc, err := json.Marshal(clone)
+	if err != nil {
+		return []byte(""), err
+	}
+
+	if tagsJSON == nil {
+		tagsJSON = make(map[string]string)
+		getMissingJSONTags(reflect.TypeOf(*c), tagsJSON)
```

Comment:
nit: I'd prefer we store this in a variable in the pkg (`var Foo = getMissingJSONTags(GettableApiAlertingConfig{})`)